### PR TITLE
Adding Husky and Lint-Staged for automatic prettifying of files

### LINKED ISSUE #203: Automatically Prettify and Lint after every commit using Husky and Lint-Staged
- **Describe the Feature Request**
Currently we're required to run `npm run format` to prettify the entire code and `npm run lint:fix` to fix the code. However, we can use Husky and lint-staged to automatically format those files which are staged into Git, making it more convenient.

- **Describe Preferred Solution**
We add Husky (which is a git hooks implementation for NodeJS projects) to execute our commands after pre-commit. We use lint-staged to filter out files which are in our staging area, and run our Prettify and ESLint commands there. This automation will massively increase code-maintenance in the future.

### PR DESCRIPTION
**Pull request by nirmalhk7**
Signed-off-by: Nirmal Khedkar <nirmalhk7@gmail.com>

- **What does this PR do?**
This PR adds Husky and lint-staged, to automatically run commands when any git hooks are run. Currently, this MR will prettify all .js files in the project.

- **Any background context you want to provide?**
I tried adding command for lint fixing as well, but when I add it to lint-staged, I get the following errors:

> .../src/demo/index.jsx
  23:19  error  Parsing error: Unexpected token =
> .../src/lib/index.jsx
  12:6  error  Parsing error: Unexpected token Props
> 
> ✖ 2 problems (2 errors, 0 warnings)

I'm not too familiar with why these errors are occurring so I've left ESLint from the lint-staged area, for now.

This will fix #203

### CODE REVIEW

**Comment by nirmalhk7** [comment]
Please review @apollonian . Thank you so much!

**Comment by apollonian** [comment]
Thanks! I'll check it out.

File: package.json [hunk 1]
```diff
@@ -68,5 +70,15 @@
...
-  "dependencies": {}
+  "dependencies": {},
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "**/*.{js,jsx}": [
+      "npx prettier --write"
```
Comment (apollonian) [inline package.json:81]: @nirmalhk7 Just one thing: Can you remove npx from here? We have prettier as a dev dependency already!

**Review by apollonian — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (nirmalhk7) [reply package.json]: Fixed.

**Review by apollonian — APPROVED** [review]
Thanks!

**Comment by apollonian** [comment]
@nirmalhk7 Could you take care of the conflicts here, please?

**Comment by nirmalhk7** [comment]
> @nirmalhk7 Could you take care of the conflicts here, please?

I'm sorry, but there arent any conflicts at all with base branch. Can you elaborate?

**Comment by apollonian** [comment]
Oh, well, I'll rebase this since now it shows commits that are already present in dev.

**Pull request closed without merging**